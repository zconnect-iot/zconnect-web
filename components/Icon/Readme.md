```jsx
<Icon name="SPANNER" size={30} color="success" />
<Icon name="PERSON" size={30} color="danger" />
<Icon name="EVENT" size={30} color="primary" />
<Icon name="DONE_ALL" size={30} color="secondary" />
```
