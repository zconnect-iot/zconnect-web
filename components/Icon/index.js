import React from 'react'
import PropTypes from 'prop-types'
import BEMHelper from 'react-bem-helper'
import ICONS from '../../assets/icons/map.json'

const classes = BEMHelper({ name: 'Icon' })

export default function Icon({ name, color, size, className, onClick }) {
  const { transform, d } = ICONS[name]
  const fill = color ? `text-${color}` : null

  return (
    <svg
      {...classes(null, name, [fill, className])}
      width={size ? `${size}px` : '100%'}
      height={size ? `${size}px` : '100%'}
      viewBox="0 0 24 24"
      onClick={onClick}
    >
      <path
        transform={transform}
        d={d}
      />
    </svg>
  )
}

Icon.propTypes = {
  name: PropTypes.string.isRequired,
  color: PropTypes.string,
  size: PropTypes.number,
  className: PropTypes.string,
  onClick: PropTypes.func,
}

Icon.defaultProps = {
  size: null,
  color: '',
  className: '',
  onClick: () => {},
}
