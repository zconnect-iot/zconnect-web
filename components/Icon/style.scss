:global(.Icon) {

  &:global(.Icon--ADD) {
    padding-top: 3px;
  }
}
