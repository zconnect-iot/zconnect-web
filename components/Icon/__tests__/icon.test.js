import React from 'react'
import Icon from '../'

describe('<Icon/>', () => {
  test('should render an Icon', () => {
    const wrapper = shallow(<Icon name="SHOPPING_CART" />)
    expect(wrapper).toMatchSnapshot()
  })
})
