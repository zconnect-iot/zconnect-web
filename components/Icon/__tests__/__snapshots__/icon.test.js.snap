// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon/> should render an Icon 1`] = `
<svg
  className="Icon Icon--SHOPPING_CART"
  height="16px"
  style={
    Object {
      "null": null,
    }
  }
  viewBox="-3 -3 24 24"
  width="16px"
>
  <path
    d="M 5.99977 16C 4.89977 16 4.00977 16.9 4.00977 18C 4.00977 19.1 4.89977 20 5.99977 20C 7.09977 20 7.99977 19.1 7.99977 18C 7.99977 16.9 7.09977 16 5.99977 16ZM 0 0L 0 2L 2 2L 5.6 9.59L 4.25 12.04C 4.09 12.32 4 12.65 4 13C 4 14.1 4.9 15 6 15L 18 15L 18 13L 6.42 13C 6.28 13 6.17 12.89 6.17 12.75L 6.2 12.63L 7.1 11L 14.55 11C 15.3 11 15.96 10.59 16.3 9.97L 19.88 3.48C 19.96 3.34 20 3.17 20 3C 20 2.45 19.55 2 19 2L 4.21 2L 3.27 0L 0 0ZM 15.9998 16C 14.8998 16 14.0098 16.9 14.0098 18C 14.0098 19.1 14.8998 20 15.9998 20C 17.0998 20 17.9998 19.1 17.9998 18C 17.9998 16.9 17.0998 16 15.9998 16Z"
  />
</svg>
`;
